scripts/simulate_agents.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import logging
+import os
+from multiprocessing import Process
+from time import sleep
+
+import wazuh_testing.tools.agent_simulator as ag
+from wazuh_testing import TCP
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+def run_agents(agents_number=1, manager_address='localhost', protocol=TCP, agent_version='v4.0.0',
+               agent_os='debian8', eps=1000, run_duration=20, active_modules=None, modules_eps=None):
+    """Run a batch of agents connected to a manager with the same parameters.
+
+    Args:
+        agents_number (int): Number of agents to run.
+        manager_address (str): Manager address (hostname or IP).
+        protocol (str): Communication protocol.
+        agent_version (str): Agents version p.e: v4.0.0
+        agent_os (str): Agents os, some examples: debian8, ubuntu18.04, mojave...
+        eps (int): Total events per second sent by each agent to the manager.
+        run_duration (int): duration in seconds of the agents run.
+        active_modules (list): list with active modules names.
+        modules_eps (list): list with eps for each active modules.
+    """
+
+    logger = logging.getLogger(f"P{os.getpid()}")
+    logger.info(f"Starting {agents_number} agents.")
+
+    active_agents, injectors = [], []
+
+    if active_modules is None:
+        active_modules = []
+
+    for _ in range(agents_number):
+        agent = ag.Agent(manager_address, "aes", os=agent_os, version=agent_version, fim_eps=eps)
+        available_modules = agent.modules.keys()
+        sending_modules = len(active_modules)
+        if 'receive_messages' in active_modules:
+            sending_modules -= 1
+        for index, module in enumerate(available_modules):
+            if module in active_modules:
+                agent.modules[module]['status'] = 'enabled'
+                if modules_eps is not None:
+                    agent.modules[module]['eps'] = modules_eps[index]
+                else:
+                    agent.modules[module]['eps'] = eps//sending_modules
+            else:
+                agent.modules[module]['status'] = 'disabled'
+                agent.modules[module]['eps'] = 0
+
+        for module in active_modules:
```

Comment:
This block should go before the second `for` loop.

Reply:
https://github.com/wazuh/wazuh-qa/pull/1148/commits/2f8cf6f1bc8896a82530968320a50c3d7c7edd84